CACHING.md

```diff
@@ -0,0 +1,143 @@
+# Cache Recipes
+
+This document focuses on gathering all the common use cases that will help the users of `actions/cache`
+- Know the use cases that can be tackled and sample workflows on how to tackle them
+- Optimise their workflows to use the cache inputs and outputs better
+
+## Actions Cache Basics
+
+The cache action works using the following set of inputs and outputs as mentioned [here](https://github.com/actions/cache#inputs). However these inputs are self explaining below are some ways in which the inputs can be used in a better way.
+
+### Keys
+
+#### Static keys
```

Comment:
rather than minor cases like static keys, focus on the most important use cases. For example,

- One of the most common use case is to use hash for lockfile as a key. This way same set of cached dependencies can be easily restored.
- Another use case could be to create different caches for different OS. In that case runner os can be used as a key
- For very short term (or isolated) use cases where cache is supposed to be short lived commit sha can be used. Similarly, for a cache which is scoped to a specific workflow run, the run id can be used.

Reply:
I've added these from line no. 37 to 58. I can update them with the descriptions that you have provided in the comment.